pypit/par/pypitpar.py

```diff
@@ -0,0 +1,2486 @@
+"""
+Defines parameter sets used to set the behavior for core pypit
+functionality.
+
+There are two main parameter set groups, those used for reducing the
+data:
+
+    - RunPar: Parameters specific to a given execution of PypIt
+
+    - ReducePar: Parameters the define how PypIt should perform the
+      reductions.  These include general parameters and the following
+      parameter subsets:
+        - OverscanPar: Methods used for the overscan subtraction.
+        - FlatFieldPar: Methods used for field-flattening.
+        - SkySubtractionPar: Methods used for sky subtraction.
+        - FlexurePar: Methods used for flexure correction.
+        - WavelengthCalibrationPar: Methods used for constructing the
+          wavelength solution.
+        - FluxCalibrationPar: Methods used for flux calibration.
+
+    - FrameGroupPar: Sets parameters that are used to group and combined
+      frames.
+        - CombineFramesPar - Parameters for combining frames
+
+    - WavelengthSolutionPar: Parameters used for constructing the wavelength
+      solution
+
+    - TraceSlitPar: Parameters for tracing slits
+        - PCAPar: PCA parameters
+
+    - TraceTiltsPar: Parameters for tracing tilts of arc lines
+
+    - TraceObjectsPar: Parameter for tracing objects
+
+    - ExtractObjectPar: Parameters for extracting 1D object spectra
+        - ManualExtractionPar: Parameters needed for manual extraction
+          of 1D spectra
+
+And those that define the parameters of the instrument used to collect
+the data:
+
+    - DetectorPar: Specifies the properties of a given detector
+
+    - InstrumentPar: Specifies the instrument used for a given
+      observation and contains the list of detectors for the camera.
+
+    - FrameFitsPar: Defines the set of fits files properties that all
+      fits files from the given instrument should have.
+
+    - FrameIDPar: Defines a set of parameters used to identify fits
+      files as a certain frame type.
+
+These are collected into the main, high-level paramter set, called
+PypitPar.
+
+.. todo::
+    - Warn the user if a default value is being used?
```

Comment:
Nah, people will accept that they are using defaults.
They will likely prefer it.

Reply:
I think it is a good idea to save a single file (or two) that contains the complete list of parameters used for the reduction. That way, whenever a user comes back to a reduction they can see exactly what parameters (including the version of PypIt) they used if they want to reproduce their results. On this note, we need to decide when to reorganise the branch structure of PypIt so that we can save versions, as @kbwestfall once suggested  :-)